packages/discovery-provider/src/tasks/entity_manager/entities/playlist.py

```diff
@@ -190,6 +198,74 @@ def update_playlist_tracks(params: ManageEntityParameters, playlist_record: Play
     )
 
 
+def update_album_price_history(
+    session: Session,
+    playlist_record: Playlist,
+    playlist_metadata: dict,
+    blocknumber: int,
+    timestamp: datetime,
+):
+    """Adds an entry in the track price history table to record the price change of a track or change of splits if necessary."""
+    new_record = None
+    is_stream_gated = playlist_metadata.get(
+        "is_stream_gated", False
+    ) and playlist_metadata.get("stream_conditions", None)
+    if is_stream_gated:
+        conditions = playlist_metadata["stream_conditions"]
+        if USDC_PURCHASE_KEY in conditions:
+            usdc_purchase = conditions[USDC_PURCHASE_KEY]
+            new_record = AlbumPriceHistory()
+            new_record.playlist_id = playlist_record.playlist_id
+            new_record.block_timestamp = timestamp
+            new_record.blocknumber = blocknumber
+            new_record.splits = {}
+            if "price" in usdc_purchase:
+                price = usdc_purchase["price"]
+                if isinstance(price, int):
+                    new_record.total_price_cents = price
+                else:
+                    raise IndexingValidationError(
+                        "Invalid type of usdc_purchase gated conditions 'price'"
+                    )
+            else:
+                raise IndexingValidationError(
+                    "Price missing from usdc_purchase gated conditions"
+                )
+
+            if "splits" in usdc_purchase:
+                splits = usdc_purchase["splits"]
+                # TODO: better validation of splits
+                if isinstance(splits, dict):
```

Comment:
Do we have any tests around this indexer? Might be nice to validate that we correctly reject invalid fields.

Reply:
adding follow-up ticket PAY-2553 to add this validation and test it